source/includes/indexes/index-code-examples.rb

```diff
@@ -0,0 +1,100 @@
+require 'mongo'
+
+# Replace the placeholders with your credentials
+uri = "<connection string>"
+
+# Sets the server_api field of the options object to Stable API version 1
+options = { server_api: { version: "1" }}
+
+# Creates a new client and connects to the server
+client = Mongo::Client.new(uri, options)
+
+database = client.use('<database name>')
+collection = database[:<collection name>]
+
+# Single Field
+# start-index-single
+collection.indexes.create_one({ <field name>: 1 })
+# end-index-single
+
+# Compound
+# start-index-compound
+collection.indexes.create_one({ <field name 1>: -1, <field name 2>: 1 })
+# end-index-compound
+
+# Multikey
+# start-index-multikey
+collection.indexes.create_one({ <field name>: 1 })
+# end-index-multikey
+
+# Geospatial
+# start-index-geospatial
+collection.indexes.create_one({ <GeoJSON field name>: '2dsphere' })
+# end-index-geospatial
+
+# Atlas Search
+
+# Create Search Index
+# start-create-search-index
+index_definition = {
+  mappings: {
+    dynamic: false,  
+    fields: { 
+      <field name>: {type: '<field type>'}
```

Comment:
We recommend adding a space at the beginning and end of hash and array literals, for readability.

```suggestion
      <field name>: { type: '<field type>' }
```